distributed/worker.py

```diff
@@ -1494,9 +1494,23 @@ async def start(self):
 
         setproctitle("dask-worker [%s]" % self.address)
 
-        await asyncio.gather(
+        plugins_msgs = await asyncio.gather(
             *(self.plugin_add(plugin=plugin) for plugin in self._pending_plugins)
         )
+        plugins_exceptions = []
+        for msg in plugins_msgs:
+            if msg["status"] != "OK":
+                exc = pickle.loads(msg["exception"].data)
```

Comment:
The pickle-unpickle here feels a little cumbersome to me. I'm not sure I love this idea, but what if you add a `catch_errors=True` to `plugin_add`, and in `start` just set `catch_errors=False`?

```diff
diff --git a/distributed/worker.py b/distributed/worker.py
index 6e2fc56b..f4e3ed03 100644
--- a/distributed/worker.py
+++ b/distributed/worker.py
@@ -1494,22 +1494,12 @@ class Worker(ServerNode):
 
         setproctitle("dask-worker [%s]" % self.address)
 
-        plugins_msgs = await asyncio.gather(
-            *(self.plugin_add(plugin=plugin) for plugin in self._pending_plugins)
+        await asyncio.gather(
+            *(
+                self.plugin_add(plugin=plugin, catch_errors=False)
+                for plugin in self._pending_plugins
+            )
         )
-        plugins_exceptions = []
-        for msg in plugins_msgs:
-            if msg["status"] != "OK":
-                exc = pickle.loads(msg["exception"].data)
-                plugins_exceptions.append(pickle.loads(msg["exception"].data))
-        if len(plugins_exceptions) >= 1:
-            if len(plugins_exceptions) > 1:
-                logger.error(
-                    "Multiple plugin exceptions raised. All exceptions will be logged, the first is raised."
-                )
-                for exc in plugins_exceptions:
-                    logger.error(repr(exc))
-            raise plugins_exceptions[0]
 
         self._pending_plugins = ()
 
@@ -3182,7 +3172,7 @@ class Worker(ServerNode):
     def run_coroutine(self, comm, function, args=(), kwargs=None, wait=True):
         return run(self, comm, function=function, args=args, kwargs=kwargs, wait=wait)
 
-    async def plugin_add(self, comm=None, plugin=None, name=None):
+    async def plugin_add(self, comm=None, plugin=None, name=None, catch_errors=True):
         with log_errors(pdb=False):
             if isinstance(plugin, bytes):
                 plugin = pickle.loads(plugin)
@@ -3204,6 +3194,8 @@ class Worker(ServerNode):
                     if isawaitable(result):
                         result = await result
                 except Exception as e:
+                    if not catch_errors:
+                        raise
                     msg = error_message(e)
                     return msg
```
If you wanted to maintain the behavior you have here where multiple plugin exceptions can be handled and logged, then you could pass `gather(..., return_exceptions=True)` and handle them similarly as here, just without the unpickling. But I'm not sure if this functionality is actually necessary.